extensions/blocks/revue/revue.php

```diff
@@ -0,0 +1,188 @@
+
+	<?php
+	return ob_get_clean();
+}
+
+/**
+ * Create the Revue subscribe button.
+ *
+ * @see https://github.com/WordPress/gutenberg/blob/015555fcdf648b13af57e08cee60bf3f3501ff63/packages/block-library/src/navigation/index.php
+ *
+ * @param array $attributes Array containing the Revue block attributes.
+ *
+ * @return string
+ */
+function jetpack_get_revue_button( $attributes ) {
+	$classes = array( 'wp-block-button__link' );
+	$styles  = array();
+
+	$has_text                    = array_key_exists( 'text', $attributes );
+	$has_class_name              = array_key_exists( 'className', $attributes );
+	$has_named_text_color        = array_key_exists( 'textColor', $attributes );
+	$has_custom_text_color       = array_key_exists( 'customTextColor', $attributes );
+	$has_named_background_color  = array_key_exists( 'backgroundColor', $attributes );
+	$has_custom_background_color = array_key_exists( 'customBackgroundColor', $attributes );
+	$has_named_gradient          = array_key_exists( 'gradient', $attributes );
+	$has_custom_gradient         = array_key_exists( 'customGradient', $attributes );
+	$has_border_radius           = array_key_exists( 'borderRadius', $attributes );
+
+	if ( $has_class_name ) {
+		$classes[] = $attributes['className'];
+	}
+
+	if ( $has_named_text_color || $has_custom_text_color ) {
+		$classes[] = 'has-text-color';
+	}
+	if ( $has_named_text_color ) {
+		$classes[] = sprintf( 'has-%s-color', $attributes['textColor'] );
+	} elseif ( $has_custom_text_color ) {
+		$styles[] = sprintf( 'color: %s;', $attributes['customTextColor'] );
+	}
+
+	if (
+		$has_named_background_color ||
+		$has_custom_background_color ||
+		$has_named_gradient ||
+		$has_custom_gradient
+	) {
+		$classes[] = 'has-background';
+	}
+	if ( $has_named_background_color && ! $has_custom_gradient ) {
+		$classes[] = sprintf( 'has-%s-background-color', $attributes['backgroundColor'] );
+	}
+	if ( $has_named_gradient ) {
+		$classes[] = sprintf( 'has-%s-gradient-background', $attributes['gradient'] );
+	} elseif ( $has_custom_gradient ) {
+		$styles[] = sprintf( 'background: %s;', $attributes['customGradient'] );
+	}
+	if (
+		$has_custom_background_color &&
+		! $has_named_background_color &&
+		! $has_named_gradient &&
+		! $has_custom_gradient
+	) {
+		$styles[] = sprintf( 'background-color: %s;', $attributes['customBackgroundColor'] );
+	}
+
+	if ( $has_border_radius ) {
+		// phpcs:ignore WordPress.PHP.StrictComparisons.LooseComparison
+		if ( 0 == $attributes['borderRadius'] ) {
+			$classes[] = 'no-border-radius';
+		} else {
+			$styles[] = sprintf( 'border-radius: %spx;', $attributes['borderRadius'] );
+		}
+	}
+
+	ob_start();
+	?>
+
+<div class="wp-block-button">
+	<button
+		class="<?php echo esc_attr( implode( ' ', $classes ) ); ?>"
+		name="member[subscribe]"
+		style="<?php echo esc_attr( implode( ' ', $styles ) ); ?>"
+		type="submit"
+	>
+		<?php echo $has_text ? esc_html( $attributes['text'] ) : esc_html__( 'Subscribe', 'jetpack' ); ?>
```

Comment:
I don't know if `esc_html` is the best thing to use here, given that one can edit the button text to add line breaks for example. Maybe we can use `wp_kses_post` instead?

```suggestion
		<?php echo $has_text ? wp_kses_post( $attributes['text'] ) : esc_html__( 'Subscribe', 'jetpack' ); ?>
```

Reply:
Fixed in f1a155d856e5a2e7ead8c9a1e57b4ce8bd709f06